crawl-ref/source/actor.cc

```diff
@@ -708,7 +708,7 @@ bool actor::can_constrict(const actor &defender, constrict_type typ) const
     if (!see_cell_no_trans(defender.pos()))
         return false;
 
-    return typ != CONSTRICT_BVC
+    return (typ != CONSTRICT_BVC && typ != CONSTRICT_ROOTS)
```

Comment:
There are three constrict types: MELEE, BVC, and ROOTS. We know this isn't MELEE because of lines 702-706, so `typ != CONSTRICT_BVC && typ != CONSTRICT_ROOTS` will never return true. This is equivalent to `return false`.

The correct fix is to delete the `typ` check. This returns the relevant logic to what it was before 57a6a0d306a.